Modules/Relationships/src/Relationships.Domain/Aggregates/Relationships/Relationship.cs

```diff
@@ -203,6 +206,42 @@ private void EnsureRevocableReactivationRequestExistsFor(IdentityAddress activeI
             throw new DomainException(DomainErrors.NoRevocableReactivationRequestExists(activeIdentity));
     }
 
+    public void Decompose(IdentityAddress activeIdentity, DeviceId activeDevice)
+    {
+        EnsureRelationshipNotDecomposed();
```

Comment:
You have to check if `activeIdentity` has already decomposed. Not if any has decomposed. Because in your current implementation, only one identity can ever decompose

Reply:
Changes made.